### ISSUE TITLE
Snippet tabstop order is meaningless.

### ISSUE THREAD

**Issue by dmahurdle** [Microsoft/vscode#6250]
- VSCode Version: 0.10.11
- OS Version: OSX 10.10.5

The [snippet documentation](https://code.visualstudio.com/Docs/customization/userdefinedsnippets) implies that $1, $2 will create ordered tabstops, but the tabstops are eventually just ordered based upon which occurred first in the template line.

For example, I'd love a snippet like this for typescript:
```json
"Typescript Import": {
		"prefix": "tsimp",
		"body": [
			"import { $2 } from '$1';",
			"$3"
		],
		"description": "Typescript import snippet"
	}
```

to default the starting tabstop position to $1, to let me define the import file. This would give me code completion on the $2, the thing I'm importing. But instead VSCode just starts at $2, then tabs to $1, then to $3.

**Comment by alexandrudima** [comment]
I think this was never implemented

**Comment by bill-mybiz** [comment]
@alexandrudima I'm on 1.2.0-insider and it isn't implemented.

**Comment by alexandrudima** [comment]
This got implemented by @csvn in #7925 and will come out with the June stable release

**Comment by dmahurdle** [comment]
Thanks so much guys! Just tried it out (have been on backend too long, finally getting back to writing some ng). Works great.
